specs/template.md

```diff
@@ -0,0 +1,118 @@
+# Spec Title
+
+## Overview
+
+*The name of the component, along with a high-level description.*
+
+### Background
+
+*Relevant historical or background information, related existing issues, etc.*
+
+### Use Cases
+
+*Primary use cases for this component.*
+  
+### Features
+
+*A list of the key features unique to this component.*
+
+### Risks and Challenges
+
+*Notable risks or challenges associated with implementing the component. Would we need to make any breaking changes to other parts of fast-dna in order to achieve this component's goals?*
+
+### Prior Art/Examples
+
+*Screenshots and/or links to existing, canonical, or exemplary implementations of the component.*
+
+---
+
+## Design
+
+*Describe the design of the component, thinking through several perspectives:*
+
+- *A customer using the component on a web page.*
+- *A developer building an app with the component and interacting through HTML/CSS/JS.*
+- *A designer customizing the component.*
+
+### API
+
+*The key elements of the component's public API surface:*
+
+- *Component Name*
+- *Props/Attrs*
+- *Methods*
+- *Events*
+
+*Consider high and low-level APIs. Attempt to design a powerful and extensible low-level API with a higher-level API for developer/designer ergonomics and simplicity.*
+
+### Anatomy and Appearance
+
+*Screenshots and/or description of the basic appearance of the component. Outline its structure with a diagram of its visual tree (shadow dom). Enumerate key areas of visual customization, such as:*
+
+- *Slot Names*
+- *Host Classes*
+- *Slotted Content/Slotted Classes*
+- *CSS Parts*
+
+---
+
+## Implementation
+
+*Important aspects of the planned implementation, with careful consideration of web standards and integration.*
+
+### States
+
+*Key component states, valid state transitions, and how interactions trigger a state transition.*
+
+### Accessibility
+
+*Consider the accessibility of the component, including:*
+
+- *Keyboard Navigation and Focus*
+- *Form Input*
+- *Use with Assistive Technology*
+  - e.g. The implications shadow dom might have on how roles are presented to the AT.
```

Comment:
Should "dom" be capitalized as an acronym?  Also, if we use AT as an abbreviation, could we first do it with "Assistive Technology" so readers know from that point on what it stands for.  I had to do a second take on it.